File: server/src/main/java/com/craftmaster/lds/fgc/user/DeviceInfo.java
package com.craftmaster.lds.fgc.user;

import java.io.Serializable;
import java.time.Instant;
import java.util.UUID;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Entity
@Accessors(chain = true)
public class DeviceInfo implements Serializable {
  private static final long serialVersionUID = 20200325L;

  private UUID deviceId;

  @Id private Instant createdAt = Instant.now();

  @Column(columnDefinition = "text")
  private String userAgent;

  @Column(columnDefinition = "text")
  private String inetAddress;

  private Instant lastLogIn;
}

File: client/src/app/welcome/device-info.ts
import { Time } from "../chat/time";
import { User } from "../auth/user";

export default interface DeviceInfo {
  createdAt: Time;
  fingerprint: string;
  browserFingerprint: string;
  userAgent: string;
  inetAddress: string;
  lastLogIn: Time;
}
